glib/src/object.rs

```diff
@@ -1272,42 +1272,77 @@ pub trait ObjectExt: ObjectType {
     /// This fails if the property does not exist, if the type of the property is different than
     /// the provided value, or if the property is not writable.
     #[doc(alias = "g_object_set_property")]
-    fn set_property<'a, N: Into<&'a str>, V: ToValue>(
+    fn try_set_property<'a, N: Into<&'a str>, V: ToValue>(
         &self,
         property_name: N,
         value: V,
     ) -> Result<(), BoolError>;
 
+    /// Similar to `try_set_property` but panics instead of failing.
+    fn set_property<'a, N: Into<&'a str>, V: ToValue>(&self, property_name: N, value: V);
+
     /// Sets the property `property_name` of the object to value `value`.
     ///
     /// This fails if the property does not exist, the type of the property is different than the
     /// provided value, or if the property is not writable.
-    fn set_property_from_value<'a, N: Into<&'a str>>(
+    fn try_set_property_from_value<'a, N: Into<&'a str>>(
         &self,
         property_name: N,
         value: &Value,
     ) -> Result<(), BoolError>;
 
+    /// Similar to `try_set_property_from_value` but panics instead of failing.
+    fn set_property_from_value<'a, N: Into<&'a str>>(&self, property_name: N, value: &Value);
+
     /// Sets multiple properties of the object at once.
     ///
     /// This does not set any properties if one or more properties don't exist, values of the wrong
     /// type are provided, or if any of the properties is not writable.
     #[doc(alias = "g_object_set")]
-    fn set_properties(&self, property_values: &[(&str, &dyn ToValue)]) -> Result<(), BoolError>;
+    fn try_set_properties(&self, property_values: &[(&str, &dyn ToValue)])
+        -> Result<(), BoolError>;
+
+    /// Similar to `try_set_properties` but panics instead of failing.
+    fn set_properties(&self, property_values: &[(&str, &dyn ToValue)]);
 
     /// Sets multiple properties of the object at once.
     ///
     /// This does not set any properties if one or more properties don't exist, values of the wrong
     /// type are provided, or if any of the properties is not writable.
-    fn set_properties_from_value(&self, property_values: &[(&str, Value)])
-        -> Result<(), BoolError>;
+    fn try_set_properties_from_value(
+        &self,
+        property_values: &[(&str, Value)],
+    ) -> Result<(), BoolError>;
+
+    /// Similar to `try_set_properties_from_value` but panics instead of failing.
+    fn set_properties_from_value(&self, property_values: &[(&str, Value)]);
+
+    /// Gets the property `property_name` of the object and cast it to the type V.
+    ///
+    /// Fails if the property doesn't exist or is not readable or of a different type than V.
+    fn try_property<'a, N: Into<&'a str>, V: FromValue<'static> + 'static>(
+        &self,
+        property_name: N,
+    ) -> Result<V, BoolError>;
+
+    /// Similar to `try_property` but panics instead of failing.
+    fn property<'a, N: Into<&'a str>, V: FromValue<'static> + 'static>(
```

Comment:
This and the others can be updated to the `for<'b> FromValue<'b>` stuff now, and make use of `Value::get_owned()`